File: YeoHun/211207/5525.swift
import Foundation

let N = readLine()!.components(separatedBy: " ").map{ Int($0)!}.first!
let M = readLine()!.components(separatedBy: " ").map{ Int($0)!}.first!
let S = readLine()!

let targetString = Array(S)
var answer = 0
var n = 0
var contFlag = false

for i in 0..<S.count - 3 {
    if contFlag == true {
        contFlag = false
        continue
    }
    let subString = String(targetString[i..<i+3])
    if subString == "IOI" {
        n += 1
        if n == N {
            answer += 1
            n -= 1
        }
        contFlag = true
    } else {
        n = 0
    }
}

print(answer)

File: YuJin/211207/5525.py
import sys
input = sys.stdin.readline

n, m, s = int(input()), int(input()), str(input())
count = 0
i, p = 0, 0

while i < m-1:
    if s[i:i+3] == 'IOI':
        p += 1
        if p == n:
            count += 1
            p -= 1
        i +=1
    else:
        p = 0
    i += 1

print(count)
